### ISSUE TITLE
Silent delay upon restart with prior db state of 24hours worth of pubnet 

### ISSUE THREAD

**Issue by sreuland** [stellar/stellar-rpc#128]
### What version are you using?
20.3.5
### What did you do?

Ran rpc for several days configured for pubnet, the on-disk db had maximum window of data allowed.
Stopped the rpc service.
Started the rpc service.

### What did you expect to see?
new logs emitted and http service is listening right after the rpc service is started.

### What did you see instead?
the rpc process is running on the o/s, but no logs emitted and no http service is bound/listening on tcp port, this state lasted for about 30minutes.

**Comment by sreuland** [comment]
I think this condition will be resolved/obsoleted by https://github.com/stellar/soroban-rpc/issues/115 which removes in-memory data storage in favor of all data on disk, @2opremio do you agree, since the delay observed here is caused by a blocking routine that copies db to memory first before it starts http service?

**Comment by 2opremio** [comment]
Yes, I agree!

**Comment by sreuland** [comment]
I don't think this bug was obsoleted and re-opening for further review. 

As of 23.0.4, there still appears to be blocking behavior that rpc performs migrations sequentially first before it starts the http service, using logs from a reported incident:

```
Oct 13 13:06:37 time="2025-10-13T13:06:37.745Z" level=info msg="starting Stellar RPC" commit=7c2b6f34c6b5f953d574e1b4f0ca0cf4ee78cf97 pid=1 version=23.0.4-7c2b6f34c6b5f953d574e1b4f0ca0cf4ee78cf97

... no log output from rpc and rpc's http server port is not up/accessible.

.. rpc was restarted a couple times, over about an 1.5 hours 

... upon this last restart
Oct 13 14:26:11  time="2025-10-13T14:26:11.992Z" level=info msg="starting Stellar RPC" commit=7c2b6f34c6b5f953d574e1b4f0ca0cf4ee78cf97 pid=1 

... no output until 20 minutes later, and then this gets reported about migrations finished, and http server is started now.

Oct 13 14:48:11  time="2025-10-13T14:48:11.335Z" level=info msg="Skipping completed migration TransactionsTable" pid=1
Oct 13 14:48:11  time="2025-10-13T14:48:11.335Z" level=info msg="Skipping completed migration EventsTable" pid=1
Oct 13 14:48:11  time="2025-10-13T14:48:11.421Z" level=info msg="Initializing in-memory store" first=59362889 last=59362939 pid=1
Oct 13 14:48:14  time="2025-10-13T14:48:14.270Z" level=info msg="Finished initializing in-memory store and applying DB data migrations" first=59241979 last=59362939 pid=1
Oct 13 14:48:14 time="2025-10-13T14:48:14.272Z" level=info msg="starting HTTP server" addr="[::]:8000" pid=1

```

**Comment by mollykarcher** [comment]
I think in this particular case, the issue was that the person experiencing the issue upgraded from a version older than 23.0.3. That version included a migration which was expected to take about an hour: https://github.com/stellar/stellar-rpc/releases/tag/v23.0.3

**Comment by sreuland** [comment]
yes, maybe if some type of recurrent log line  like `migrations ongoing ...` could be emitted from rpc once every minute or so during migration it would provide some feedback to operators that process isn't hung. the only prior log output is when the migration starts, which in this case was missed and not captured.
